.travis.yml

```diff
@@ -40,24 +40,30 @@ script:
 
 jobs:
   include:
-# Travis-ci doesn't know how to install Maria 10.1
-#   - php: 7.1
-#     env: DB=mysql MARIADB=10.1
-#     addons:
-#       mariadb: 10.1
-#   - php: 7.2
-#     env: DB=mysql MARIADB=10.1
-#     addons:
-#       mariadb: 10.1
-#   - php: 7.3
-#     env: DB=mysql MARIADB=10.1
-#     addons:
-#       mariadb: 10.1
-#   - php: 7.4
-#     env: DB=mysql MARIADB=10.1
-#     addons:
-#       mariadb: 10.1
-#       TODO add maria 5.5, it is not EOL yet.
+    - php: 7.1
+      dist: trusty
+      env: DB=mysql MARIADB=10.1
+      services:
+      addons:
+        mariadb: 10.1
+    - php: 7.2
```

Comment:
Please only always add min, max. There is no point in adding php in between.

Reply:
-1. We have edge cases all the time.

Reply:
Not in that language spectrum. PHP min+max solve all those issues, always.
We have years of experience proving this.

Reply:
```$ diff 71 72
8,9c8,9
< phpdocumentor/reflection-docblock  4.3.4   
< phpdocumentor/type-resolver        1.0.1   
---
> phpdocumentor/reflection-docblock  5.1.0   
> phpdocumentor/type-resolver        1.2.0   
33,34c33,35
< symfony/filesystem                 v4.4.10 
< symfony/finder                     v4.4.10 
---
> symfony/deprecation-contracts      v2.1.2  
> symfony/filesystem                 v5.1.2  
> symfony/finder                     v5.1.2  
39c40
< symfony/service-contracts          v1.1.8  
---
> symfony/service-contracts          v2.1.2  
41c42
< symfony/translation-contracts      v1.1.7  
---
> symfony/translation-contracts      v2.1.2  
43c44
< symfony/yaml                       v4.4.10 
---
> symfony/yaml                       v5.1.2  
```

This will get worse as we start supporting libraries that need to support php8 and start dropping 7.1, 7.2, etc. The dependencies we heavily leverage here are changing all the time and result in actually different versions, which creates actually different bugs for us.

Reply:
Once PHP8 becomes an issue, we can discuss that
for now, in 7.1-7.4, testing 7.2 and 7.3 is pointless as it provides no additional info here. none.

Reply:
to be clear: 7.1 and 7.4 as min and max showcase the end of spectrums for PHP dependency, the language itself has a clear way of introducing and deprecating/removing functionality. so the ones in between never show anything that isn't shown by the outer ones.

Reply:
I would like us to spend resources, and our time, on the other things still open, that matter.
Running 20 from 40+ tests in the suite for nothing at all is a waste of resources, and slows everyone down.

Reply:
The packages installed for 71 and 72 and 73 are not the same because of package constraints against php. We need to run everything.

In getting travis to pass I encountered non-stop silly edgecases that were hidden somewhere like one exact package range that was only installable under some other co-conditions of composer or travis.

This is not a showcase, it is a CI matrix to find exactly that kind of issue. I see 0 value in removing builds that test exact configurations.

Reply:
Yes, of course. But the min and max range cover the extremes already.
The packages in between are not relevant for normal use cases.
Even if you find a glitch in the 0.0001% of cases, that doesnt warrant a 50%+ increase in job runs.
There is no benefit here.